pkg/capi/util/clusterclass-variables.ts

```
@@ -0,0 +1,223 @@
+
+    const mockSchema = { id, resourceFields: {} };
+
+    schemas.push(mockSchema);
+    for (const key in properties) {
+      // make a uid for each type
+      const subtypeId = randomStr();
+      const def = properties[key];
+
+      if (SIMPLE_TYPES.includes(def.type)) {
+        mockSchema.resourceFields[key] = def;
+      } else if (def.type === 'array') {
+        if (SIMPLE_TYPES.includes(def.items.type)) {
+          mockSchema.resourceFields[key] = { type: 'array', subtype: def.items.type };
+        } else {
+        // heh
+          const subSchemaSubSchemas = makeSchemas(properties[key], subtypeId);
+
+          schemas.push(...subSchemaSubSchemas);
+          mockSchema.resourceFields[key] = { type: 'array', subtype: subtypeId };
+        }
+      } else if (def.additionalProperties) {
+        if (SIMPLE_TYPES.includes(def.additionalProperties.type)) {
+          mockSchema.resourceFields[key] = { type: 'map', subtype: def.additionalProperties.type };
+        } else {
+          const subSchemaSubSchemas = makeSchemas(properties[key], subtypeId);
+
+          schemas.push(...subSchemaSubSchemas);
+          mockSchema.resourceFields[key] = { type: 'map', subtype: subtypeId };
+        }
+      } else {
+        const subSchemaSubSchemas = makeSchemas(properties[key], subtypeId);
+
+        schemas.push(...subSchemaSubSchemas);
+
+        mockSchema.resourceFields[key] = { type: subtypeId };
+      }
+    }
+  }
+
+  return schemas;
+};
+
+/**
+ *
+ * @param openSchema object - openAPIV3Schema property of a clusterclass variable definition
+ * @param data existing variable configuration
+ * @returns string - yaml preview of fields defined in the openAPIV3Schema
+ */
+export const makeYamlPlaceholders = function(openSchema, data = {}) {
+  const schemas = makeSchemas(openSchema);
+  let out;
+
+  try {
+    out = createYaml(schemas, 'ccvariable', data, false);
+  } catch (err) {
+    console.error(err);
```

Let's either swallow this error or show it to the user